demos/demo1-chat-heads/Demo.jsx

```diff
@@ -1,40 +1,189 @@
+    }
+    if (this.state.stiffness !== nextState.stiffness) {
+      return true;
+    }
+    if (this.state.damping !== nextState.damping) {
+      return true;
+    }
+    if (this.state.precision !== nextState.damping) {
+      return true;
+    }
+  
+    return false;
   },
 
   handleMouseMove({pageX: x, pageY: y}) {
-    this.setState({x, y});
+    const height = this.state.headerHeight
+    const chatHeadRadiusPx = 20
+    // chat heads avoid header area
+    if(y > height + chatHeadRadiusPx){
+      this.setState({x, y: y - height});
+    }
   },
 
   handleTouchMove({touches}) {
     this.handleMouseMove(touches[0]);
   },
 
+  handleStiffnessChange(event) {
+    this.setState({stiffness: event.target.value});
+  },
+
+  handlePrecisionChange(event) {
+    if(!isNaN(event.target.value)){
+      this.setState({precision: event.target.value});
+    }
+  },
+
+  handleDampingChange(event) {
+    this.setState({damping: event.target.value});
+  },
+
+  handleStiffnessParamChange(event) {
+    this.setState({stiffnessBehaviour: event.newValue});
+  },
+
+  handleDampingParamChange(event) {
+    this.setState({dampingBehaviour: event.newValue});
+  },
+  
   getStyles(prevStyles) {
     // `prevStyles` is the interpolated value of the last tick
+    let stiffness = this.state.stiffness
+    let damping = this.state.damping
+    let precision = this.state.precision
+
     const endValue = prevStyles.map((_, i) => {
+      if (this.state.stiffnessBehaviour === 'linear'){
+        stiffness = this.state.stiffness * i
+      }
+      if (this.state.stiffnessBehaviour === 'inverseLinear'){
+        stiffness = this.state.stiffness / i
+      }
+      if (this.state.dampingBehaviour === 'linear'){
+        damping = this.state.damping * i
+      }
+      if (this.state.dampingBehaviour === 'inverseLinear'){
+        damping = this.state.damping / i
+      }
       return i === 0
         ? this.state
-        : {
-            x: spring(prevStyles[i - 1].x, presets.gentle),
-            y: spring(prevStyles[i - 1].y, presets.gentle),
+        : {  
+            x: spring(prevStyles[i - 1].x, {stiffness, damping, precision}),
+            y: spring(prevStyles[i - 1].y, {stiffness, damping, precision}),
           };
     });
     return endValue;
   },
 
-  render() {
+  render() {   
+    const options = [
+        {
+            description: 'Constant',
+            value: 'constant'
+        },
+        {
+            description: 'Linear',
+            value: 'linear'
+        },
+        {
+            description: 'Inverse Linear',
+            value: 'invertedLinear'
+        },
+    ];
+
     return (
+      <div>
+        <div className="header"  ref={element => this.divRef = element}>
+          <div className="parent-container">
+            <div className="dropdown-container">
+              <div className=".mdl-selectfield">
+                <label>Stiffness Behaviour</label>
+                <Dropdown className='Dropdown' 
```

Comment:
Ok thanks. I'll update the pull request and make sure that the bare minimum logic is there to add the spring customisations.